nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/PostgreSQLBackend.java

```diff
@@ -85,8 +86,9 @@ public Map<String, POSTGRESQL_COLUMN_TYPES> listOfFields(String attrPersistence,
                         else if (subAttribute.getAttrValue() instanceof Number){
                             if (aggregation.replace(encodedSubAttrName, POSTGRESQL_COLUMN_TYPES.NUMERIC) == null)
                                 aggregation.putIfAbsent(encodedSubAttrName, POSTGRESQL_COLUMN_TYPES.NUMERIC);
-                        }
-                        else aggregation.putIfAbsent(encodedSubAttrName, POSTGRESQL_COLUMN_TYPES.TEXT);
+                        } else if(subAttribute.getAttrValue() instanceof JSONObject){
```

Comment:
A sub-attribute being a JSON object does not imply it is a geometry. What's more, I'm wondering if we can have a geometry as a sub-attribute (as an attribute, yes, but as a sub-attribute?)?